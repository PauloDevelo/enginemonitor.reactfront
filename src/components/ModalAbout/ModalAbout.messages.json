{
  "about": {
    "defaultMessage": "Made with love by Paul Torruella",
    "id": "aboutMsg"
  },
  "ok": {
    "defaultMessage": "OK",
    "id": "ok"
  },
  "title": {
    "defaultMessage": "Equipment Maintenance",
    "id": "aboutTitle"
  }
}
