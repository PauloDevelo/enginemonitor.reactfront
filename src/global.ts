import packageJson from '../package.json';

export interface Global{
  getAppVersion(): string;
}

class EquipmentMaintenanceGlobal implements Global {
  getAppVersion = () => packageJson.version
}

const appVersion:Global = new EquipmentMaintenanceGlobal();

export default appVersion;
