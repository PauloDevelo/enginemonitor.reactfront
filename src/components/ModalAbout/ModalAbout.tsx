import React from 'react';
import {
  Button, Modal, ModalHeader, ModalBody, ModalFooter, Form, Media,
} from 'reactstrap';

import { FormattedMessage, defineMessages } from 'react-intl';

import global from '../../global';

import jsonMessages from './ModalAbout.messages.json';

const aboutMsg = defineMessages(jsonMessages);

type Props = {
visible: boolean;
className?: string;
toggle: ()=>void;
}

const ModalAbout = ({ visible, className, toggle }: Props) => (
  <Modal isOpen={visible} toggle={toggle} className={className} fade>
    <ModalHeader toggle={toggle}>
      <Media left>
        <Media object src="./images/engine48.png" alt="Equipment Maintenance icon" />
      </Media>
      <Media body>
        <Media heading>
          <FormattedMessage {...aboutMsg.title} />
&nbsp;v
          {global.getAppVersion()}
        </Media>
        <FormattedMessage {...aboutMsg.about} />
      </Media>
    </ModalHeader>
    <ModalBody>
      {visible && (<Form id="aboutForm" />)}
    </ModalBody>
    <ModalFooter>
      <Button type="submit" form="aboutForm" color="success" onClick={toggle}><FormattedMessage {...aboutMsg.ok} /></Button>
    </ModalFooter>
  </Modal>
);

export default ModalAbout;
