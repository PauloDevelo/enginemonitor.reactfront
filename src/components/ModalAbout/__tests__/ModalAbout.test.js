
import React from 'react';
import { mount } from 'enzyme';

import localforage from 'localforage';
import global from '../../../global';

// eslint-disable-next-line no-unused-vars
import ignoredMessages from '../../../testHelpers/MockConsole';
import ModalAbout from '../ModalAbout';

jest.mock('localforage');
jest.mock('../../../global');

describe('Modal About', () => {
  beforeAll(() => {
    ignoredMessages.length = 0;
    ignoredMessages.push('test was not wrapped in act(...)');
    ignoredMessages.push('Please update the following components: ReactImageLightbox');
    ignoredMessages.push('[React Intl] Could not find required `intl` object.');
  });

  beforeEach(() => {

  });

  afterEach(() => {

  });

  it('should render a modal about', async () => {
    // Arrange
    global.getAppVersion.mockImplementation(() => '2.9.12');
    const toggle = jest.fn();

    // Act
    const modalAbout = mount(<ModalAbout visible toggle={toggle} />);

    // Assert
    expect(modalAbout).toMatchSnapshot();
  });
});
