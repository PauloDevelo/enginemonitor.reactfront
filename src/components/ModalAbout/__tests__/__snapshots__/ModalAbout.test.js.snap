// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal About should render a modal about 1`] = `
<ModalAbout
  toggle={[MockFunction]}
  visible={true}
>
  <Modal
    autoFocus={true}
    backdrop={true}
    backdropTransition={
      Object {
        "mountOnEnter": true,
        "timeout": 150,
      }
    }
    centered={false}
    fade={true}
    isOpen={true}
    keyboard={true}
    modalTransition={
      Object {
        "timeout": 300,
      }
    }
    onClosed={[Function]}
    onOpened={[Function]}
    returnFocusAfterClose={true}
    role="dialog"
    scrollable={false}
    toggle={[MockFunction]}
    unmountOnClose={true}
    zIndex={1050}
  >
    <Portal
      node={
        <div
          style="position: relative; z-index: 1050; display: block;"
          tabindex="-1"
        >
          <div
            class=""
          >
            <div
              class="modal fade"
              role="dialog"
              style="display: block;"
              tabindex="-1"
            >
              <div
                class="modal-dialog"
                role="document"
              >
                <div
                  class="modal-content"
                >
                  <div
                    class="modal-header"
                  >
                    <h5
                      class="modal-title"
                    >
                      <div
                        class="media-left"
                      >
                        <img
                          alt="Equipment Maintenance icon"
                          class="media-object"
                          src="./images/engine48.png"
                        />
                      </div>
                      <div
                        class="media-body"
                      >
                        <h4
                          class="media-heading"
                        >
                          Equipment Maintenance
                           v
                          2.9.12
                        </h4>
                        Made with love by Paul Torruella
                      </div>
                    </h5>
                    <button
                      aria-label="Close"
                      class="close"
                      type="button"
                    >
                      <span
                        aria-hidden="true"
                      >
                        ×
                      </span>
                    </button>
                  </div>
                  <div
                    class="modal-body"
                  >
                    <form
                      class=""
                      id="aboutForm"
                    />
                  </div>
                  <div
                    class="modal-footer"
                  >
                    <button
                      class="btn btn-success"
                      form="aboutForm"
                      type="submit"
                    >
                      OK
                    </button>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="modal-backdrop fade"
            />
          </div>
        </div>
      }
    >
      <Portal
        containerInfo={
          <div
            style="position: relative; z-index: 1050; display: block;"
            tabindex="-1"
          >
            <div
              class=""
            >
              <div
                class="modal fade"
                role="dialog"
                style="display: block;"
                tabindex="-1"
              >
                <div
                  class="modal-dialog"
                  role="document"
                >
                  <div
                    class="modal-content"
                  >
                    <div
                      class="modal-header"
                    >
                      <h5
                        class="modal-title"
                      >
                        <div
                          class="media-left"
                        >
                          <img
                            alt="Equipment Maintenance icon"
                            class="media-object"
                            src="./images/engine48.png"
                          />
                        </div>
                        <div
                          class="media-body"
                        >
                          <h4
                            class="media-heading"
                          >
                            Equipment Maintenance
                             v
                            2.9.12
                          </h4>
                          Made with love by Paul Torruella
                        </div>
                      </h5>
                      <button
                        aria-label="Close"
                        class="close"
                        type="button"
                      >
                        <span
                          aria-hidden="true"
                        >
                          ×
                        </span>
                      </button>
                    </div>
                    <div
                      class="modal-body"
                    >
                      <form
                        class=""
                        id="aboutForm"
                      />
                    </div>
                    <div
                      class="modal-footer"
                    >
                      <button
                        class="btn btn-success"
                        form="aboutForm"
                        type="submit"
                      >
                        OK
                      </button>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="modal-backdrop fade"
              />
            </div>
          </div>
        }
      >
        <div
          className=""
        >
          <Fade
            appear={true}
            baseClass="fade"
            baseClassActive="show"
            className="modal"
            enter={true}
            exit={true}
            in={true}
            mountOnEnter={false}
            onClick={[Function]}
            onEnter={[Function]}
            onEntered={[Function]}
            onEntering={[Function]}
            onExit={[Function]}
            onExited={[Function]}
            onExiting={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            role="dialog"
            style={
              Object {
                "display": "block",
              }
            }
            tabIndex="-1"
            tag="div"
            timeout={300}
            unmountOnExit={false}
          >
            <Transition
              appear={true}
              enter={true}
              exit={true}
              in={true}
              mountOnEnter={false}
              onEnter={[Function]}
              onEntered={[Function]}
              onEntering={[Function]}
              onExit={[Function]}
              onExited={[Function]}
              onExiting={[Function]}
              timeout={300}
              unmountOnExit={false}
            >
              <div
                className="modal fade"
                onClick={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                role="dialog"
                style={
                  Object {
                    "display": "block",
                  }
                }
                tabIndex="-1"
              >
                <div
                  className="modal-dialog"
                  role="document"
                >
                  <div
                    className="modal-content"
                  >
                    <ModalHeader
                      charCode={215}
                      closeAriaLabel="Close"
                      tag="h5"
                      toggle={[MockFunction]}
                      wrapTag="div"
                    >
                      <div
                        className="modal-header"
                      >
                        <h5
                          className="modal-title"
                        >
                          <Media
                            left={true}
                          >
                            <div
                              className="media-left"
                            >
                              <Media
                                alt="Equipment Maintenance icon"
                                object={true}
                                src="./images/engine48.png"
                              >
                                <img
                                  alt="Equipment Maintenance icon"
                                  className="media-object"
                                  src="./images/engine48.png"
                                />
                              </Media>
                            </div>
                          </Media>
                          <Media
                            body={true}
                          >
                            <div
                              className="media-body"
                            >
                              <Media
                                heading={true}
                              >
                                <h4
                                  className="media-heading"
                                >
                                  <FormattedMessage
                                    defaultMessage="Equipment Maintenance"
                                    id="aboutTitle"
                                    values={Object {}}
                                  >
                                    Equipment Maintenance
                                  </FormattedMessage>
                                   v
                                  2.9.12
                                </h4>
                              </Media>
                              <FormattedMessage
                                defaultMessage="Made with love by Paul Torruella"
                                id="aboutMsg"
                                values={Object {}}
                              >
                                Made with love by Paul Torruella
                              </FormattedMessage>
                            </div>
                          </Media>
                        </h5>
                        <button
                          aria-label="Close"
                          className="close"
                          onClick={[MockFunction]}
                          type="button"
                        >
                          <span
                            aria-hidden="true"
                          >
                            ×
                          </span>
                        </button>
                      </div>
                    </ModalHeader>
                    <ModalBody
                      tag="div"
                    >
                      <div
                        className="modal-body"
                      >
                        <Form
                          id="aboutForm"
                          tag="form"
                        >
                          <form
                            className=""
                            id="aboutForm"
                          />
                        </Form>
                      </div>
                    </ModalBody>
                    <ModalFooter
                      tag="div"
                    >
                      <div
                        className="modal-footer"
                      >
                        <Button
                          color="success"
                          form="aboutForm"
                          onClick={[MockFunction]}
                          tag="button"
                          type="submit"
                        >
                          <button
                            aria-label={null}
                            className="btn btn-success"
                            form="aboutForm"
                            onClick={[Function]}
                            type="submit"
                          >
                            <FormattedMessage
                              defaultMessage="OK"
                              id="ok"
                              values={Object {}}
                            >
                              OK
                            </FormattedMessage>
                          </button>
                        </Button>
                      </div>
                    </ModalFooter>
                  </div>
                </div>
              </div>
            </Transition>
          </Fade>
          <Fade
            appear={true}
            baseClass="fade"
            baseClassActive="show"
            className="modal-backdrop"
            enter={true}
            exit={true}
            in={true}
            mountOnEnter={true}
            onEnter={[Function]}
            onEntered={[Function]}
            onEntering={[Function]}
            onExit={[Function]}
            onExited={[Function]}
            onExiting={[Function]}
            tag="div"
            timeout={150}
            unmountOnExit={false}
          >
            <Transition
              appear={true}
              enter={true}
              exit={true}
              in={true}
              mountOnEnter={true}
              onEnter={[Function]}
              onEntered={[Function]}
              onEntering={[Function]}
              onExit={[Function]}
              onExited={[Function]}
              onExiting={[Function]}
              timeout={150}
              unmountOnExit={false}
            >
              <div
                className="modal-backdrop fade"
              />
            </Transition>
          </Fade>
        </div>
      </Portal>
    </Portal>
  </Modal>
</ModalAbout>
`;
